arch/x86/crc32_fold_pclmulqdq.c

```diff
@@ -24,12 +25,17 @@
 #include <smmintrin.h> // _mm_extract_epi32
 
 #include "x86_features.h"
+#include "cpu_features.h"
 
 #include "../../crc32_fold.h"
+#include "../../crc32_p.h"
 
 #ifdef X86_VPCLMULQDQ_CRC
 extern size_t fold_16_vpclmulqdq(__m128i *xmm_crc0, __m128i *xmm_crc1,
     __m128i *xmm_crc2, __m128i *xmm_crc3, uint8_t *dst, const uint8_t *src, size_t len);
+extern size_t fold_16_vpclmulqdq_nocp(__m128i *xmm_crc0, __m128i *xmm_crc1,
```

Comment:
Why not just use `fold_16_vpclmulqdq` and check for `dst == NULL`?

Reply:
That's a good point, given that we unconditionally get to do _1_ load here (guaranteed by the initial size check), we only need a single branch at the beginning.  Have to run out for a bit but this will be a quick change.

Reply:
Ah, at second glance I also removed the simultaneous copy from that version.  So, in a loop I'm avoid a bunch of if dst != NULL jumps.